README.md

```diff
@@ -1,55 +1,385 @@
-| [MPSC Queue](src_lockfree/mpsc_queue.mli)                    | `Single_consumer_queue` (same)                | A multi-producer, single-consumer, thread-safe queue without support for cancellation. This makes a gooddata structure for a scheduler's run queue. It is used in [Eio](https://github.com/ocaml-multicore/eio).                                                               | It is a single consumer version of the queue described in [Implementing lock-free queues](https://people.cs.pitt.edu/~jacklange/teaching/cs2510-f12/papers/implementing_lock_free.pdf).                      |
+```ocaml
+# test ();;
+- : int * int list = (15, [1; 1; 1; 1; 1; 2; 1; 1; 1; 2; 2; 2; 2; 1; 2])
+```
 
-## Usage
+and as expected it is not working. The interleaving scenario describes
+previously can still happen. Note that, though it is not observable here, this
+is still better then the previous implementation as we are now race-free (see
+[here](#a-note-about-races-in-ocaml) for a quick note about races in OCaml).
 
-`Saturn` can be installed from `opam`: `opam install saturn`. Sample usage of
-`Work_stealing_deque` is illustrated below.
+As a matter of fact, writting a queue with `push` and `pop` functions that are
+both atomic and parallelism-safe is not as easy as it mays
+
+With `Saturn`,
 
 ```ocaml
-module Ws_deque = Work_stealing_deque.M
+let work_saturn id barrier q () =
+  Barrier.await barrier;
+  for i = 0 to 9 do
+    Saturn.Queue.push q id
+  done
 
-let q = Ws_deque.create ()
+let test_saturn () =
+  let barrier = Barrier.create 2 in
+  let q = Saturn.Queue.create () in
+  let d1 = Domain.spawn (work_saturn 1 barrier q) in
+  let d2 = Domain.spawn (work_saturn 2 barrier q) in
+  Domain.join d1;
+  Domain.join d2;
+  let rec pop_all acc =
+    match Saturn.Queue.pop q with
+    | None -> List.rev acc
+    | Some elt -> pop_all (elt :: acc)
+  in
+  let res = pop_all [] in
+  (List.length res, res)
+```
 
-let () = Ws_deque.push q 100
+Let's run it:
 
-let () = assert (Ws_deque.pop q = 100)
+```ocaml
+# test_saturn ();;
+- : int * int list =
+(20, [2; 2; 1; 2; 2; 2; 2; 2; 2; 1; 2; 1; 2; 1; 1; 1; 1; 1; 1; 1])
 ```
 
+### A note about races in OCaml
+
+Because of the great properties of OCaml 5 memory model (see the
+[OCaml manual](https://v2.ocaml.org/manual/parallelism.html#s%3Apar_mm_easy) for
+more details), not a lot can go worst here: at least, not data corruption or
+segmentation fault like it could happen in other languages.
```

Comment:
```suggestion
more details), not a lot can go worst here. At least, data corruption or
segmentation fault won't happen like it can in other languages.
```

Can go "worst" or can go "wrong"?

Reply:
I prefer "not a lot can go wrong".

Reply:
Agree with KC here